pageserver/src/tenant/mgr.rs

```diff
@@ -495,17 +495,8 @@ pub async fn init_tenant_mgr(
         let mut location_conf = match location_conf {
             Ok(l) => l,
             Err(e) => {
-                warn!(tenant_id=%tenant_shard_id.tenant_id, shard_id=%tenant_shard_id.shard_slug(), "Marking tenant broken, failed to {e:#}");
-
-                tenants.insert(
-                    tenant_shard_id,
-                    TenantSlot::Attached(Tenant::create_broken_tenant(
-                        conf,
-                        tenant_shard_id,
-                        resources.remote_storage.clone(),
-                        format!("{}", e),
-                    )),
-                );
+                // This should only happen in the case of a serialization bug or critical local I/O error: we cannot load this tenant
+                error!(tenant_id=%tenant_shard_id.tenant_id, shard_id=%tenant_shard_id.shard_slug(), "Failed to load tenant config, failed to {e:#}");
                 continue;
```

Comment:
I broadly agree, but it's orthogonal to this PR: the change here is just that we used to mark the tenant broken, now we don't load it at all.  Being stricktier and panickier makes sense, but one could have made the point about  the previous code.

Reply:
Having the broken tenant occupy the slot in the map prevented follow-up errors in later code because it would bail out early. 

Now the slot will be free.

Does it matter (especially for `/location_config`)?

It does `mkdir -p` (`create_dir_all`), so, the tenant will spawn on whatever the on-disk state is.

Going through what happens on `attach`, I can't spot any obvious problems, perhaps it was a bigger concern in past days when local state was authoritative.

So, I guess feel free to resolve. If you think the proposed items (1) to (3) above are worth doing, please create a follow-up ticket and link it here.

Reply:
>Going through what happens on attach, I can't spot any obvious problems, perhaps it was a bigger concern in past days when local state was authoritative.

Right, the attach is always verifying everything local against remote state (and over-writing config with whatever came in the location_config request), so there shouldn't be any room for mistakes.)

I'll do a followup that purges the anyhow::Result stuff from here to remove ambiguity.